nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,187 @@
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    threshold: None, 'auto' or float
+        Passed to CanICA. If None, no thresholding is applied. If 'auto',
+        then we apply a thresholding that will keep the n_voxels,
+        more intense voxels across all the maps, n_voxels being the number
+        of voxels in a brain volume. A float value indicates the
+        ratio of voxels to keep (2. means keeping 2 x n_voxels voxels).
+
+    n_init: int, optional
+        The number of times the fastICA algorithm is restarted
+
+    random_state: int or RandomState
+        Pseudo number generator state used for random sampling.
+
+    target_affine: 3x3 or 4x4 matrix, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape: 3-tuple of integers, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    low_pass: None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    high_pass: None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    t_r: float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs: integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed
+
+    References
+    ----------
+    * G. Varoquaux et al. "A group model for stable multi-subject ICA on
+      fMRI datasets", NeuroImage Vol 51 (2010), p. 288-299
+
+    * G. Varoquaux et al. "ICA-based sparse features recovery from fMRI
+      datasets", IEEE ISBI 2010, p. 1177
+    """
+
+    def __init__(self, mask=None, n_components=20,
+                 smoothing_fwhm=6,
+                 standardize=True,
+                 random_state=0,
+                 target_affine=None, target_shape=None,
+                 low_pass=None, high_pass=None, t_r=None,
+                 alpha=1,
+                 batch_size=10,
+                 n_iter=1000,
+                 # Common options
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, verbose=0,
+                 ):
+        CanICA.__init__(self,
```

Comment:
As discussed, I'd like to be able to bootstrap the dictionary learning with KMeans. Not a big deal, I think you should just add a `init_maps` parameter and call CanICA only if it is not provided.